apps/prairielearn/src/middlewares/authzWorkspace.ts

```diff
@@ -60,6 +60,9 @@ export default asyncHandler(async (req, res, next) => {
       await authzHasCoursePreviewOrInstanceView(req, res);
       await selectAndAuthzInstructorQuestion(req, res);
     }
+  } else if (res.locals.share_publicly || res.locals.share_source_publicly) {
+    // This workspace is not part of a course instance and the question is publicly accessible
+    // Therefore, no authorization is needed
   } else if (res.locals.course_id) {
```

Comment:
From discussion on Slack:

We don't want to have _some_ amount of auth here, just because we don't want any person to be able to access any workspace and do things.

It might be as simple as, for public previews, we only allowing people to access workspaces created by their own user id. But we should think about it more to make sure it makes sense.

Reply:
For public question previews, I can't figure out any reason you would ever need to access someone else's workspace. If you need to view someone else's work, it would be through a question in a course instance.

Reply:
I went ahead and implemented only having it be accessible if the variant user id matches the authn_user id.

I think this is reasonable because it will make things work as expected for now without permitting things we don't want.

In the future, if we have some use case for people accessing a workspace from a public question preview created by someone else, then we could revisit it.